src/sentry/integrations/gitlab/integration.py

```diff
@@ -0,0 +1,210 @@
+        IntegrationFeatures.ISSUE_BASIC,
+    ])
+
+    setup_dialog_config = {
+        'width': 1030,
+        'height': 1000,
+    }
+
+    def _make_identity_pipeline_view(self):
+        """
+        Make the nested identity provider view. It is important that this view is
+        not constructed until we reach this step and the
+        ``oauth_config_information`` is available in the pipeline state. This
+        method should be late bound into the pipeline vies.
+        """
+        identity_pipeline_config = dict(
+            oauth_scopes=(),
+            redirect_url=absolute_uri('/extensions/gitlab/setup/'),
+            verify_ssl=False,
+            **self.pipeline.fetch_state('oauth_config_information')
+        )
+
+        return NestedPipelineView(
+            bind_key='identity',
+            provider_key='gitlab',
+            pipeline_cls=IdentityProviderPipeline,
+            config=identity_pipeline_config,
+        )
+
+    def get_oauth_data(self, payload):
+        data = {'access_token': payload['access_token']}
+
+        if 'expires_in' in payload:
+            data['expires'] = int(time()) + int(payload['expires_in'])
+        if 'refresh_token' in payload:
+            data['refresh_token'] = payload['refresh_token']
+        if 'token_type' in payload:
+            data['token_type'] = payload['token_type']
+
+        return data
+
+    def get_user_info(self, access_token, installation_data):
+        session = http.build_session()
+        resp = session.get(
+            'https://{}/api/v4/user'.format(installation_data['url']),
+            headers={
+                'Accept': 'application/json',
+                'Authorization': 'Bearer %s' % access_token,
+            },
+        )
+
+        resp.raise_for_status()
+        return resp.json()
+
+    def get_group_info(self, access_token, installation_data):
+        session = http.build_session()
+        resp = session.get(
+            'https://{}/api/v4/groups/{}'.format(
+                installation_data['url'], installation_data['group']),
+            headers={
+                'Accept': 'application/json',
+                'Authorization': 'Bearer %s' % access_token,
+            },
+        )
+
+        resp.raise_for_status()
+        return resp.json()
+
+    def get_pipeline_views(self):
+        return [InstallationConfigView(), lambda: self._make_identity_pipeline_view()]
+
+    def build_integration(self, state):
+        data = state['identity']['data']
+        oauth_data = self.get_oauth_data(data)
+        user = self.get_user_info(data['access_token'], state['installation_data'])
+        group = self.get_group_info(data['access_token'], state['installation_data'])
+        scopes = sorted(GitlabIdentityProvider.oauth_scopes)
+        base_url = state['installation_data']['url']
+
+        integration = {
+            'name': group['name'],
+            # probs should add on base url to make it unique like ghe
```

Comment:
Yes! Is this something we can do right now? Or is it something we have to wait until we have access to the gitlab instance james is working on?